api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -134,19 +135,20 @@ public Workspace get(String ns, String firecloudName) {
                 fcWorkspace -> fcWorkspace));
   }
 
+  /**
+   * Swagger Java codegen does not handle the WorkspaceACL model correctly; it returns a GSON map
+   * instead. Run this through a typed Gson conversion process to parse into the desired type.
+   */
+  private static Map<String, WorkspaceAccessEntry> extractAclResponse(WorkspaceACL aclResp) {
+    Type accessEntryType = new TypeToken<Map<String, WorkspaceAccessEntry>>() {}.getType();
+    Gson gson = new Gson();
+    return gson.fromJson(gson.toJson(aclResp.getAcl(), accessEntryType), accessEntryType);
+  }
+
   @Override
   public Map<String, WorkspaceAccessEntry> getFirecloudWorkspaceAcls(
       String workspaceNamespace, String firecloudName) {
-    WorkspaceACL firecloudWorkspaceAcls =
-        fireCloudService.getWorkspaceAcl(workspaceNamespace, firecloudName);
-    Map<String, Object> aclsMap = (Map) firecloudWorkspaceAcls.getAcl();
-    Map<String, WorkspaceAccessEntry> userToAcl = new HashMap<>();
-    for (Map.Entry<String, Object> entry : aclsMap.entrySet()) {
-      WorkspaceAccessEntry acl =
-          new Gson().fromJson(entry.getValue().toString(), WorkspaceAccessEntry.class);
-      userToAcl.put(entry.getKey(), acl);
-    }
-    return userToAcl;
+    return extractAclResponse(fireCloudService.getWorkspaceAcl(workspaceNamespace, firecloudName));
```

Comment:
looks good except why make this its own method?  seems like you could just put the above code into `getFirecloudWorkspaceAcls` to consolidate, and add a comment for clarity

Reply:
Done.